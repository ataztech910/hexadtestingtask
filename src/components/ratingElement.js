import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';

export const RatingElement = props => (
  <>
    <ListItemAvatar>
      <Avatar
        alt={`Image for Â°${props.name}`}
        src={`https://picsum.photos/80/80?image=${props.imgId}`}
      />
    </ListItemAvatar>
    <ListItemText className="elementRating" primary={props.name} secondary={`Rating is: ${props.rating}`} />
    <ListItemSecondaryAction>
      <IconButton
        color="primary"
        component="span"
        className="increaseRating"
        onClick={(event) => {
          event.preventDefault();
          props.setRating({ id: props.id, action: 1 });
        }}
      >
        <Icon>thumb_up</Icon>
      </IconButton>
      <IconButton
        color="primary"
        className="decreaseRating"
        component="span"
        onClick={(event) => {
          event.preventDefault();
          props.setRating({ id: props.id, action: -1 });
        }}
      >
        <Icon>thumb_down</Icon>
      </IconButton>
    </ListItemSecondaryAction>
  </>
);
export default RatingElement;
