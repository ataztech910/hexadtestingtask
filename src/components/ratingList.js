import React from 'react';
import RatingElementContainer from '../containers/RatingElementConatainer';

export const RatingList = list => (
  <ul className="rating-list">
    {list.fetch.list.ratings && list.fetch.list.ratings
      ? list.fetch.list.ratings.map(element => <RatingElementContainer key={`element-${element.id}`} {...element} />)
      : 'No elements, pitty =('}
  </ul>
);
export default RatingList;
