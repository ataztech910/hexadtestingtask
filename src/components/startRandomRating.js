import React from 'react';
import Button from '@material-ui/core/Button';

const StartRandomRating = props => (
  <Button
    variant="contained"
    color="primary"
    type="button"
    onClick={(event) => {
      event.preventDefault();
      props.startRandomProcess();
    }}
  >
    {props.state.startProcess ? 'Stop random rates' : 'Start random rates' }
  </Button>
);

export default StartRandomRating;
