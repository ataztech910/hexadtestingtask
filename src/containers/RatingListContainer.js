import { connect } from 'react-redux';
import { RatingList } from '../components/ratingList';
import { getAll } from '../actions/ratings';

const mapStateToProps = (state) => {
  const fetch = getAll(state);
  return { fetch };
};


export default connect(
  mapStateToProps, getAll,
)(RatingList);
