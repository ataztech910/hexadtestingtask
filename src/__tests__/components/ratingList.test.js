import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ListItem from '@material-ui/core/ListItem';
import { RatingList } from '../../components/ratingList';
import RatingElementContainer from '../../containers/RatingElementConatainer';
import RatingListContainer from '../../containers/RatingListContainer';

configure({ adapter: new Adapter() });

const mockStore = configureMockStore();
const store = mockStore({});

describe('RatingList check', () => {
  const props = {
    fetch: {
      list: {
        ratings: [{ name: 'Test', rating: 1 }],
      },
    },
  };
  const listElement = shallow(<RatingList {...props} />);
  const elementContainer = shallow(<RatingElementContainer store={store} />).dive();
  const elementListContainer = shallow(<RatingListContainer store={store} />).dive();

  it('renders properly', () => {
    expect(listElement).toMatchSnapshot();
  });

  it('renders container properly', () => {
    expect(elementContainer).toMatchSnapshot();
    expect(elementListContainer).toMatchSnapshot();
  });

  it('renders list properly', () => {
    expect(listElement.find('.element')).toBeDefined();
  });
  it('List item mounted', () => {
    expect(listElement.contains(<ListItem {...props} />)).toBeDefined();
    expect(listElement.find('.element')).toHaveLength(props.fetch.list.ratings.length);
  });
});
