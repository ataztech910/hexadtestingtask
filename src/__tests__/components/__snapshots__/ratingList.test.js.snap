// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingList check renders container properly 1`] = `ShallowWrapper {}`;

exports[`RatingList check renders container properly 2`] = `ShallowWrapper {}`;

exports[`RatingList check renders properly 1`] = `ShallowWrapper {}`;
